source/shared/cpp/ObjectModel/BaseElement.cpp

```diff
@@ -53,44 +54,44 @@ namespace AdaptiveSharedNamespace
     void BaseElement::SetAdditionalProperties(Json::Value const& value) { m_additionalProperties = value; }
 
     // Given a map of what our host provides, determine if this element's requirements are satisfied.
-    bool BaseElement::MeetsRequirements(const std::unordered_map<std::string, std::string>& hostProvides) const
+    bool BaseElement::MeetsRequirements(const AdaptiveSharedNamespace::FeatureRegistration& featureRegistration) const
     {
         for (const auto& requirement : m_requires)
         {
             // special case for adaptive cards version
             const auto& requirementName = requirement.first;
             const auto& requirementVersion = requirement.second;
-            if (requirementName == "adaptiveCards")
+            const auto& featureVersion = featureRegistration.GetFeatureVersion(requirementName);
+            if (featureVersion.empty())
             {
-                static const SemanticVersion currentAdaptiveCardsVersion{"1.2"};
-                if (currentAdaptiveCardsVersion > requirementVersion)
-                {
-                    return false;
-                }
+                // host doesn't provide this requirement
+                return false;
             }
             else
             {
-                const auto& provides = hostProvides.find(requirementName);
-                if (provides == hostProvides.end())
+                // host provides this requirement, but does it provide an acceptible version?
+                const SemanticVersion providesVersion{featureVersion};
+                if (providesVersion < requirementVersion)
                 {
-                    // host doesn't provide this requirement
+                    // host's provided version is too low
                     return false;
                 }
-                else
-                {
-                    // host provides this requirement, but does it provide an acceptible version?
-                    const SemanticVersion providesVersion{provides->second};
-                    if (providesVersion < requirementVersion)
-                    {
-                        // host's provided version is too low
-                        return false;
-                    }
-                }
             }
         }
+
         return true;
     }
 
+    std::shared_ptr<std::unordered_map<std::string, AdaptiveSharedNamespace::SemanticVersion>> BaseElement::GetRequirements() const
+    {
+        return std::make_shared<std::unordered_map<std::string, AdaptiveSharedNamespace::SemanticVersion>>(m_requires);
+    }
+
+    void BaseElement::SetRequirements(const std::unordered_map<std::string, AdaptiveSharedNamespace::SemanticVersion>& requirements)
```

Comment:
>SetRequirements [](start = 22, length = 15)

Our other collection types just have a Get, and then the callers add things to it (like GetBody() or GetActions()). I realize those are vectors and this is a map, but i'm wondering if we should use the same approach here